carbon-design-system/carbon-addons-iot-react
# 1766 hotspot tooltip tab

Closes #1766

**Summary**
Created a new HotspotTooltipTab component.

![image](https://user-images.githubusercontent.com/5167091/98940216-f36ebc80-24ea-11eb-99f6-57e9dfb75664.png)


**Change List (commits, features, bugs, etc)**
- built for external state management of form fields and dropdowns
- made customizable with custom icons and colors using override-pattern
- linked icons and color dropdowns
- hides all fields when the infoMessage is present
- added delete icon button
- added callbacks for onChange and on Delete
- added stories
- added unit tests

**Acceptance Test (how to verify the PR)**
1. Go to https://deploy-preview-1792--carbon-addons-iot-react.netlify.app?path=/story/watson-iot-experimental-hotspoteditormodal-hotspoteditortooltiptab--with-state-in-story
2. Interact with the controls and verify that the state is being managed in the story
3. check that color and icons are linked
4. check that actions are using callbacks

1. Go to https://deploy-preview-1792--carbon-addons-iot-react.netlify.app?path=/story/watson-iot-experimental-hotspoteditormodal-hotspoteditortooltiptab--with-preset-values-and-custom-colors
2. Verify that the icons, color and texts are preset

1. Go to https://deploy-preview-1792--carbon-addons-iot-react.netlify.app?path=/story/watson-iot-experimental-hotspoteditormodal-hotspoteditortooltiptab--with-info-message
2. Verify that the info message is showing and nothing else
3. Remove the message using the knob and make sure the form is visible again
